ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1293,6 +1304,19 @@ private void prepareForSending() {
       }
     }
 
+    /**
+     * Resolve the reserved metadata id to be used for data chunks.
+     * For simple blobs or metadata chunks this will be {@code null}. For the data chunks of composite blobs, the reserved
+     * metadata id of the metadata chunk will be used.
+     * @return the blob id to use as the reserved metadata id. Can be {@code null}.
+     */
+    private String resolveReservedMetadataId() {
+      if (!routerConfig.routerReservedMetadataEnabled || isSimpleBlob || isMetadataChunk()) {
```

Comment:
I thought we pass the reserved meta data id to the properties no matter we enable the flag or not?
```
!routerConfig.routerReservedMetadataEnabled
```

Reply:
We are not serializing the reserved metadata to disk yet, but we are still assigning reserved metadata for Post operations. For example, if its a chunked upload, the signed url created to do the upload will have a reserved metadata id embedded in it.  `validateReservedMetadataChunkId` will check against that embedded reserved metadata id. It doesn't need to get the reserved metadata id from the ambry storage.

Also note that unless `routerConfig.routerReservedMetadataEnabled` is true, no exception will be actually thrown out of the `MetadataChunk` constructor.